common/network_util.h

```diff
@@ -16,11 +16,20 @@
 
 #include <string>
 
+#include "folly/SocketAddress.h"
+
+
 namespace common {
 
 /*
  * Get the local IP address from eth0.
  */
 const std::string& getLocalIPAddress();
 
+/**
+ * Get a string representation of the provide socket address if it is IPv4 or IPv6,
+ * otherwise "unknown_addr" is returned.
+ */
+std::string getNetworkAddressStr(const folly::SocketAddress&) noexcept;
```

Comment:
Two questions:
1. could you provide some education links for this c++ syntax, that we ignore the variable name, and only indicate the types. 
2. does the socketAddress get updated somewhere else, which make you return a copy instead of a reference?

Reply:
1. it should be perfectly fine to not include a variable name in the function declaration to indicate the types. Just a random search: http://www.cplusplus.com/forum/general/28012/. It's even legal to do this in the actual 
2. this is a utilization function to transform any input SocketAddress into a string, so the original SocketAddress isn't stored anywhere and will be destroyed once the function returns. If we were to return a reference, we would need to  allocate the SocketAddress in the heap (probably with `static`), something we should probably for a temporary variable I think